src/main/java/seedu/duke/command/task/ViewTaskCommand.java

```diff
@@ -0,0 +1,54 @@
+package seedu.duke.command.task;
+
+import seedu.duke.model.project.Project;
+import seedu.duke.model.project.ProjectList;
+import seedu.duke.model.task.Task;
+import seedu.duke.ui.Messages;
+import seedu.duke.ui.Ui;
+
+import java.util.Hashtable;
+
+public class ViewTaskCommand extends TaskCommand {
+    private final ProjectList projectListManager;
+    private Project proj;
+
+    public ViewTaskCommand(Hashtable<String, String> parameters, ProjectList projectListManager) {
+        super(parameters);
+        this.projectListManager = projectListManager;
+    }
+
+    public void execute() {
+        assert !projectListManager.isEmpty() : "No project found!\n";
+        if (projectListManager.isEmpty()) {
+            Ui.showError("Please create a project first.");
+            return;
+        }
+
+        try {
+            Project proj = projectListManager.getProject();
+            if (parameters.isEmpty()) {
+                Ui.showError("Missing parameters.");
+            }
+            Ui.showToUserLn("The details of the tasks are as follows: ");
+            for (int i = 0; i < parameters.size(); i++) {
+                Task task;
+                try {
```

Comment:
I think no need to nest another `try-catch`, you can just put another catch for `NumberFormatException` in the other `try-catch`.